lib/host/wasi_nn/wasinnfunc.cpp

```diff
@@ -6,46 +6,434 @@
+  [[maybe_unused]] uint32_t *Context =
+      MemInst->getPointer<uint32_t *>(ContextPtr, 1);
+
+  if (Ctx.GraphBackends.size() <= GraphId) {
+    logger->error("init_execution_context: Graph Id does not exist");
+    return -1;
+  }
+
+  if (Ctx.GraphBackends[GraphId] == Ctx.BackendsMapping.at("OpenVINO")) {
+#ifdef WASMEDGE_WASINN_BUILD_OPENVINO
+    IEStatusCode Status;
+    OpenVINOSession *Session = new OpenVINOSession();
+    Session->exe_network = Ctx.OpenVINOExecutions[GraphId];
+    Session->network = Ctx.OpenVINONetworks[GraphId];
+    if (Session->exe_network == nullptr || Session->network == nullptr) {
+      logger->error("Model for Graph:{} is empty!", GraphId);
+      return -1;
+    }
+    Status = ie_exec_network_create_infer_request(Session->exe_network,
+                                                  &(Session->infer_request));
+
+    if (Status != OK || Session->infer_request == nullptr) {
+      logger->error("Unable to create openvino session");
+      return -1;
+    }
+
+    Ctx.ExecutionsNum = Ctx.OpenVINOInfers.size();
+    Ctx.OpenVINOInfers.push_back(Session);
+    Ctx.GraphContextBackends.push_back(GraphId);
+    *Context = Ctx.ExecutionsNum;
+
+    return 0;
+#else
+    logger->error("OpenVINO backend is not built. define "
+                  "-DWASMEDGE_WASINN_BUILD_OPENVINO "
+                  "to build it.");
+#endif
+  } else {
+    logger->error("Current backend is not supported.");
+  }
+  return -1;
 }
 
-Expect<uint32_t> WasiNNSetInput::body(Runtime::Instance::MemoryInstance *MemInst
-                                      [[maybe_unused]],
-                                      uint32_t Context [[maybe_unused]],
-                                      uint32_t Index [[maybe_unused]],
-                                      uint32_t TensorPtr [[maybe_unused]]) {
-  return 0;
+Expect<uint32_t>
+WasiNNSetInput::body(Runtime::Instance::MemoryInstance *MemInst,
+                     uint32_t Context, uint32_t Index [[maybe_unused]],
+                     uint32_t TensorPtr [[maybe_unused]]) {
+  if (MemInst == nullptr) {
+    return Unexpect(ErrCode::ExecutionFailed);
+  }
+
+  auto logger = spdlog::get("wasi-nn");
+  if (Ctx.GraphContextBackends.size() <= Context) {
+    logger->error("set_input: Execution Context does not exist");
+    return -1;
+  }
+  if (Ctx.GraphContextBackends[Context] == Ctx.BackendsMapping.at("OpenVINO")) {
+#ifdef WASMEDGE_WASINN_BUILD_OPENVINO
+    IEStatusCode Status;
+    OpenVINOSession *Session = Ctx.OpenVINOInfers[Context];
+    ie_network_t *Network = Session->network;
+    ie_infer_request_t *InferRequest = Session->infer_request;
+    ie_blob_t *InputBlob = NULL;
+    ie_blob_buffer_t BlobCBuffer;
+    int BlobSize;
+    char *InputName = nullptr;
+
+    if (Network == nullptr || InferRequest == nullptr) {
+      logger->error("The founded openvino session is empty");
+      return -1;
+    }
+
+    uint32_t *Tensor = MemInst->getPointer<uint32_t *>(TensorPtr, 1);
```

Comment:
How long should the `Tensor` array be? 
You should give the correct size here, not `1`. The `getPointer` API will check the memory boundary, and return `nullptr` if exceeded the boundary.